kafka/consumer/fetcher.py

```diff
@@ -664,19 +665,34 @@ def _create_fetch_requests(self):
                 log.debug("Adding fetch request for partition %s at offset %d",
                           partition, position)
 
-        if self.config['api_version'] >= (0, 10):
+        if self.config['api_version'] >= (0, 10, 1):
+            version = 3
+        elif self.config['api_version'] >= (0, 10):
             version = 2
         elif self.config['api_version'] == (0, 9):
             version = 1
         else:
             version = 0
         requests = {}
         for node_id, partition_data in six.iteritems(fetchable):
-            requests[node_id] = FetchRequest[version](
-                -1, # replica_id
-                self.config['fetch_max_wait_ms'],
-                self.config['fetch_min_bytes'],
-                partition_data.items())
+            if version < 3:
+                requests[node_id] = FetchRequest[version](
+                    -1,  # replica_id
+                    self.config['fetch_max_wait_ms'],
+                    self.config['fetch_min_bytes'],
+                    partition_data.items())
+            else:
+                # As of version == 3 partitions will be returned in order as
+                # they are requested, so to avoid starvation with
+                # `fetch_max_bytes` option we need this shuffle
+                partition_data = list(partition_data.items())
+                random.shuffle(partition_data)
```

Comment:
A note to this shuffle. As python dicts and sets are not ordered, currently it will work properly without this shuffle (it will basically be shuffled by _create_fetch_requests, as it uses sets and dicts), but I still prefere it here, as:

* Python3.6 has ordered dicts, which can make the distribution way less equal
* While dicts are not retain insert order, they preserve key order by hashes, which can (probably) result in unequal consumption.

Please correct me if I'm wrong here. My tests run OK even without this shuffle, so I'm a bit concerned.